QueueBeanFactory

mtwharmby opened the pull request:
Adds new factory to create queue atoms/beans for the queue service. This is the first part of the DB-QueueService integration.

gerring commented on org.eclipse.scanning.api/src/org/eclipse/scanning/api/event/queues/models/arguments/QueueListVariable.java:
```diff
@@ -41,7 +41,8 @@ protected V processArg(Integer parameter) {
 			return list.evaluate().get(parameter);
 		} catch (IndexOutOfBoundsException iEOBEx) {
-			throw new QueueModelException("No value at index "+parameter+". ", iEOBEx);
+			return null; //FIXME!!!!
```
Why? Comment?

gerring commented on org.eclipse.scanning.api/src/org/eclipse/scanning/api/event/queues/models/arguments/QueueTableVariable.java:
```diff
@@ -42,7 +42,7 @@ public QueueTableVariable(IQueueValue<A> arg, IQueueValue<Map<A, V>> table) {
 	protected V processArg(A parameter) {
 		V tableValue = table.evaluate().get(parameter);
-		if (tableValue == null) throw new QueueModelException("No value for "+parameter+" in table");
+//		if (tableValue == null) throw new QueueModelException("No value for "+parameter+" in table"); //FIXME
```
Why?  Comment?

gerring commented on org.eclipse.scanning.api/src/org/eclipse/scanning/api/event/queues/models/arguments/QueueValue.java:
```diff
@@ -32,7 +32,7 @@ public QueueValue(String name, V value) {
 	@Override
 	public V evaluate() {
-		if (value == null) throw new QueueModelException("Value was never initialised");
+//		if (value == null) throw new QueueModelException("Value was never initialised"); //FIXME
```
Warum?

gerring commented on org.eclipse.scanning.test/src/org/eclipse/scanning/test/event/queues/QueueBeanFactoryTest.java:
```diff
@@ -0,0 +1,123 @@
+		PositionerAtom detXAtom = new PositionerAtom("setDetX", "dummy", 225);
+		detXAtom.setName("Set detX to 10");
+		qbf.registerAtom(detXAtom);
+//TODO?		atomReg = qbf.getQueueAtomRegister();
```
Why what is this?

gerring commented on org.eclipse.scanning.test/src/org/eclipse/scanning/test/event/queues/QueueBeanFactoryTest.java:
```diff
@@ -0,0 +1,123 @@
+			qbf.getQueueAtom("mvDum");
+			fail("Fetching unregistered atom did not throw an exception");
+		} catch (QueueModelException qme) {
+			//expected
```
For what reason? Does it need a logger.info(...)

mtwharmby replied:
I'm testing that an exception is thrown when the supplied string doesn't exist in the QueueBeanFactory registry. If the exception *is not* thrown, then something has gone wrong - hence the fail() in the try block. There's no point reporting that the exception was caught, unless you want a log overflowing with reports.

gerring commented:
It might be worth doing a face to face review with Emilio on this.

mtwharmby commented:
@gerring I've addressed all your comments. I'll see if I can find Emilio this afternoon - is he up to speed or do you just want me to talk through with him what the design is?

mtwharmby commented:
@gerring Emilio isn't in today and I would really like to get this merged in so I can get on with the next part of this and also test it on the beamline at the end of this week. I'm happy to keep the branch alive, so I can talk through the basic QueueBeanFactory with him.

mtwharmby commented:
@gerring Can you check my changes please? I think I addressed everything - if not let me know what I missed. Ta!
